domain/domain.go

```diff
@@ -860,7 +860,6 @@ func (do *Domain) UpdateTableStatsLoop(ctx sessionctx.Context) error {
 		return nil
 	}
 	owner := do.newStatsOwner()
-	do.wg.Add(1)
```

Comment:
Will it make leaktest fail? See https://github.com/pingcap/tidb/pull/9849

Reply:
I run `make leak`, and it pass.
I guess that in the unit test environment, there is never a huge table(analyze cost to long) and no tikv offline(block on network retry), so the worker goroutine exit quickly when the `exit` channel is closed https://github.com/pingcap/tidb/pull/10147/files#diff-46dccda7ce1de403b0cda3709f1ce5ceR917

Reply:
Should we add it back?

Reply:
This is the core change of the PR
If we wait for stats worker, tidb can't exit... and that maybe a disaster in some scene like rolling update.

Reply:
Does `autoAnalyzeWorker` have the same issue?